src/v2/client/startLoginFlow.js

```diff
@@ -0,0 +1,45 @@
+/*!
+ * Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* global Promise */
+import Errors from 'util/Errors';
+import { interact } from './interact';
+import { introspect } from './introspect';
+
+const handleProxyIdxResponse = async (settings) => {
+  return Promise.resolve({
+    rawIdxState: settings.get('proxyIdxResponse'),
+    context: settings.get('proxyIdxResponse'),
+    neededToProceed: [],
+  });
+};
+
+export async function startLoginFlow (settings) {
+  // Return a preset response
+  if (settings.get('proxyIdxResponse')) {
+    return handleProxyIdxResponse(settings);
+  }
+
+  // Use stateToken
+  const stateHandle = settings.get('stateToken');
+  if (stateHandle) {
+    return introspect(settings);
+  }
+  
+  // Use or acquire interactionHandle
+  if (settings.get('useInteractionCodeFlow')) {
+    return interact(settings);
+  }
+
+  throw new Errors.ConfigError('Set "useInteractionCodeFlow" to true in configuration to enable the '+ 
```

Comment:
This isn't documented in the README anywhere. What is our plan for adding OIE configuration options to the primary README?

Reply:
This feature is currently in EA. There is a separate migration guide which is distributed to users who have this feature enabled. When the feature is GA it is expected to be fully documented online and in the README.